packages/expo-maps/ios/AppleMapsViewiOS18.swift

```diff
@@ -1,17 +1,36 @@
 import SwiftUI
 import MapKit
+import ExpoModulesCore
 
 @available(iOS 18.0, *)
 struct AppleMapsViewiOS18: View, AppleMapsViewProtocol {
   @EnvironmentObject var props: AppleMapsViewProps
-  @ObservedObject private var state = AppleMapsViewState()
+  @ObservedObject private var state = AppleMapsViewiOS18State()
 
   func setCameraPosition(config: CameraPosition?) {
     withAnimation {
       state.mapCameraPosition = config.map(convertToMapCamera) ?? .userLocation(fallback: state.mapCameraPosition)
     }
   }
 
+  func openLookAround(coordinate: Coordinate) async throws {
+    if state.lookAroundScene != nil {
+      throw LookAroundAlreadyPresentedException()
+    }
+
+    let scene = try await getLookAroundScene(from: CLLocationCoordinate2D(
+      latitude: coordinate.latitude,
+      longitude: coordinate.longitude
+    ))
+    
+    if (scene == nil) {
```

Comment:
@alanjhughes It is still necessary to do this check, because in cases where the scene is unavailable for the location, the request will still pass, but return a nil scene. I guess the underlying call only throws if there's a network/server issue, not a "data" issue. So we should throw here as well. However, I noticed that if I throw here, my JS code doesn't enter the `catch` clause of try/catch. With promises it worked just fine. Any clue on why this would happen?

Reply:
I also tried doing

```
AsyncFunction("openLookAroundAsync") { (view: AppleMapsViewWrapper, coordinates: Coordinate, promise: Promise) in
        Task {
          do {
            try await view.openLookAround(coordinate: coordinates)
            promise.resolve(nil)
          } catch {
            promise.reject(error)
          }
        }
      }
```
It enters the catch block with the correct exception, but its like it doesn't reach JS as it never enters the catch block in JS, even though I strictly rejected the promise

Reply:
This will be resolved in #36431. There was an issue with `ConcurrentFunctionDefinition`.

Reply:
Okay great to hear - will re-test as soon as it is merged